File: python/test/test_scan_by_key.py
import ThrustRTC as trtc

ctx = trtc.Context()

d_keys = trtc.device_vector_from_list(ctx, [0, 0, 0, 1, 1, 2, 3, 3, 3, 3], 'int32_t')

d_values = trtc.device_vector_from_list(ctx, [1, 1, 1, 1, 1, 1, 1, 1, 1, 1 ], 'int32_t')
trtc.Inclusive_Scan_By_Key(ctx, d_keys, d_values, d_values)
print (d_values.to_host())

d_values = trtc.device_vector_from_list(ctx, [1, 1, 1, 1, 1, 1, 1, 1, 1, 1 ], 'int32_t')
trtc.Exclusive_Scan_By_Key(ctx, d_keys, d_values, d_values)
print (d_values.to_host())

d_values = trtc.device_vector_from_list(ctx, [1, 1, 1, 1, 1, 1, 1, 1, 1, 1 ], 'int32_t')
trtc.Exclusive_Scan_By_Key(ctx, d_keys, d_values, d_values, trtc.DVInt32(5))
print (d_values.to_host())

File: java/test/test_scan_by_key.java
import java.util.*;
import JThrustRTC.*;

public class test_scan_by_key
{
	public static void main(String[] args) 
	{
		{
			DVVector d_keys = new DVVector(new int[] { 0, 0, 0, 1, 1, 2, 3, 3, 3, 3 });
			DVVector d_values = new DVVector(new int[] { 1, 1, 1, 1, 1, 1, 1, 1, 1, 1 });
			TRTC.Inclusive_Scan_By_Key(d_keys, d_values, d_values);
			System.out.println(Arrays.toString((int[])d_values.to_host()));
		}

		{
			DVVector d_keys = new DVVector(new int[] { 0, 0, 0, 1, 1, 2, 3, 3, 3, 3 });
			DVVector d_values = new DVVector(new int[] { 1, 1, 1, 1, 1, 1, 1, 1, 1, 1 });
			TRTC.Exclusive_Scan_By_Key(d_keys, d_values, d_values);
			System.out.println(Arrays.toString((int[])d_values.to_host()));
		}

		{
			DVVector d_keys = new DVVector(new int[] { 0, 0, 0, 1, 1, 2, 3, 3, 3, 3 });
			DVVector d_values = new DVVector(new int[] { 1, 1, 1, 1, 1, 1, 1, 1, 1, 1 });
			TRTC.Exclusive_Scan_By_Key(d_keys, d_values, d_values, new DVInt32(5));
			System.out.println(Arrays.toString((int[])d_values.to_host()));
		}
	}
}
